com.unity.visualeffectgraph/Editor/ShaderGraph/VFXShaderGraphParticleOutput.cs

```diff
@@ -880,6 +868,25 @@ public override void EndCompilation()
                             }
                         }
 
+                        if (graphCode.requirements.requiresScreenPosition
+                            || graphCode.requirements.requiresNDCPosition
+                            || graphCode.requirements.requiresPixelPosition)
+                        {
+                            callSG.builder.AppendLine("{");
+                            if (graphCode.requirements.requiresNDCPosition || graphCode.requirements.requiresPixelPosition)
+                                callSG.builder.AppendLine("float2 localPixelPosition = i.VFX_VARYING_POSCS.xy;");
+                            if (graphCode.requirements.requiresNDCPosition)
+                                callSG.builder.AppendLine("float2 localNDCPosition = localPixelPosition.xy / _ScreenParams.w;");
+
+                            if (graphCode.requirements.requiresScreenPosition)
+                                callSG.builder.AppendLine("INSG.ScreenPosition = ComputeScreenPos(localPixelPosition, _ProjectionParams.x);");
```

Comment:
This seems incorrect. computeScreenPos expects HClip pos as input and you're passing float2 pixel pos. It should not even compile. Do we have test coverage for this?

Reply:
There is a special definition of this in `ShaderGraphLibrary/Functions.hlsl#L20`
https://github.com/Unity-Technologies/Graphics/blob/0de4625e8b5604e3e573b4b681b84460ef77d791/com.unity.shadergraph/ShaderGraphLibrary/Functions.hlsl#L20
This code path is only used with shaderGraph and it's equivalent to this, in URP:
https://github.com/Unity-Technologies/Graphics/blob/75f4b421fa774d43000d703f2bbf05d7a1ca6606/com.unity.render-pipelines.universal/Editor/ShaderGraph/Templates/SharedCode.template.hlsl#L56
And in HDRP:
https://github.com/Unity-Technologies/Graphics/blob/75f4b421fa774d43000d703f2bbf05d7a1ca6606/com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/Pixel.template.hlsl#L39

Reply:
So screenpos is still requiring worldpos in SG then?

Reply:
> Do we have test coverage for this?
We have two test which covers this.
1) SampleScene:
![image](https://user-images.githubusercontent.com/22494706/134647772-23043608-2d49-4bc8-9f54-6bc2b6d81eff.png)
This one is run on HDRP & URP.

2) ShadergraphUnlit
![image](https://user-images.githubusercontent.com/22494706/134647997-b3d3b80f-5838-416d-8a97-b9489c9cb748.png)
This one is HDRP only.

Reply:
Oh but it's float2 vs float4 😶
It should at least generate a warning.

Reply:
But do we really test ScreenPosition in PS? How could the test pass if it's the case?

Reply:
Indeed, I double check, I though "ScreenPosition" was used in "Default" Mode, but actually, it's correspond to the "Raw" value.
![image](https://user-images.githubusercontent.com/22494706/134664703-9852a7ea-9f95-4f62-8a56-a17cd9f22b34.png)
I'm fixing this case 🔧

Reply:
Resolved at 1db0279e862a495577f27d9a9f7e79bdedc61048 (moving back to ScreenPosition computed from WorldPos as before)